Bio/Phylo/TreeConstruction.py

```diff
@@ -692,63 +692,79 @@ def nj(self, distance_matrix):
         min_i = 0
         min_j = 0
         inner_count = 0
-        while len(dm) > 2:
-            # calculate nodeDist
-            for i in range(0, len(dm)):
-                node_dist[i] = 0
-                for j in range(0, len(dm)):
-                    node_dist[i] += dm[i, j]
-                node_dist[i] = node_dist[i] / (len(dm) - 2)
-
-            # find minimum distance pair
-            min_dist = dm[1, 0] - node_dist[1] - node_dist[0]
-            min_i = 0
-            min_j = 1
-            for i in range(1, len(dm)):
-                for j in range(0, i):
-                    temp = dm[i, j] - node_dist[i] - node_dist[j]
-                    if min_dist > temp:
-                        min_dist = temp
-                        min_i = i
-                        min_j = j
-            # create clade
+        # special case for Minimum Alignment Matrices
+        if len(dm) == 2:
+            # minimum distance will always be [1,0]
+            min_i = 1
+            min_j = 0
             clade1 = clades[min_i]
             clade2 = clades[min_j]
-            inner_count += 1
-            inner_clade = BaseTree.Clade(None, "Inner" + str(inner_count))
+            clade1.branch_length = dm[min_i, min_j] / 2.0
+            clade2.branch_length = dm[min_i, min_j] - clade1.branch_length
+            inner_clade = BaseTree.Clade(None, "Inner")
             inner_clade.clades.append(clade1)
             inner_clade.clades.append(clade2)
-            # assign branch length
-            clade1.branch_length = (dm[min_i, min_j] + node_dist[min_i] -
-                                    node_dist[min_j]) / 2.0
-            clade2.branch_length = dm[min_i, min_j] - clade1.branch_length
-
-            # update node list
-            clades[min_j] = inner_clade
-            del clades[min_i]
-
-            # rebuild distance matrix,
-            # set the distances of new node at the index of min_j
-            for k in range(0, len(dm)):
-                if k != min_i and k != min_j:
-                    dm[min_j, k] = (dm[min_i, k] + dm[min_j, k] -
-                                    dm[min_i, min_j]) / 2.0
-
-            dm.names[min_j] = "Inner" + str(inner_count)
-            del dm[min_i]
-
-        # set the last clade as one of the child of the inner_clade
-        root = None
-        if clades[0] == inner_clade:
-            clades[0].branch_length = 0
-            clades[1].branch_length = dm[1, 0]
-            clades[0].clades.append(clades[1])
+            clades[0] = inner_clade
             root = clades[0]
         else:
-            clades[0].branch_length = dm[1, 0]
-            clades[1].branch_length = 0
-            clades[1].clades.append(clades[0])
-            root = clades[1]
+            while len(dm) > 2:
```

Comment:
I think rather than indenting all the old code one level using:

```
# special case for Minimum Alignment Matrices
if len(dm) == 2:
    ... # new code
else:
    while len(dm) > 2:
     ... # old code
```

you could leave it as:

```
if len(dm) == 2:
    ... # new code
while len(dm) > 2:
    ... # old code
```

This should make the diff much smaller, and be easier to review.